src/Psalm/Internal/Analyzer/Statements/Expression/Call/StaticMethod/ExistingAtomicStaticCallAnalyzer.php

```diff
@@ -613,4 +622,29 @@ private static function getMethodReturnType(
 
         return $return_type_candidate;
     }
+
+    /**
+     * Dumb way to determine whether a type contains "static" somewhere inside.
+     */
+    private static function hasStaticInType(Type\TypeNode $type): bool
+    {
+        assert($type instanceof Atomic || $type instanceof Union);
+        $union_parts = $type instanceof Union
+            ? $type->getAtomicTypes()
+            : [ $type ];
+
+        foreach ($union_parts as $atomic_type) {
+            if ($atomic_type instanceof TNamedObject && $atomic_type->value === 'static') {
+                return true;
+            }
+
+            foreach ($atomic_type->getChildNodes() as $child_type) {
+                if (self::hasStaticInType($child_type)) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
```

Comment:
This could be simplified further to just the part inside the loop, something like:

```php
if ($type instanceof TNamedObject && $value === 'static') {
    return true;
}

foreach ($type->getChildNodes() as $child_type) {
    if (self::hasStaticInType($child_type)) {
        return true;
    }
}

return false;
```

That way there's also no need to assert `Union` or `Atomic`, which helps if I ever manage to add an `Intersection` type.